### PR TITLE
feat(provider): persist per-view view-state identity and durable viewStates

### PR DESCRIPTION
**Pull request by easonLiangWorldedtech**
Part of the vps2 durable per-view state series — tracked in easonLiangWorldedtech/Zoo-Code#41 (cross-repo: this PR is standalone against upstream/main @ 0d937c050).

Issue (created at PR-open time): #1547

## What
At the base commit `ClineProvider` has no per-view state identity: every view shares the same global mode / profile / apiConfiguration keys, nothing is persisted per view, `postMessageToWebview` awaits an ack a remounted page never sends, and reset / history-restore writes leak across views. This PR lands the durable per-view core (fix unit F1a, 1/3): per-view identity, the `viewStates` persistence pipeline, and the view-local state buffer. The `getState()` merging of hydrated per-view values and the webview-side identity / launch wiring land in the follow-ups (F1b / F1c).

## Design decisions
- Per-view identity: `viewId` = renderContext plus a monotonic counter (unique per instance for its lifetime). `viewStateId` is the stable durable key (registered by the webview launch flow in F1c); `rekeyPersistedViewStateEntry` moves the temporary-id entry to the stable id, stable id winning on a collision.
- Durable writes go through a serialized write queue (`savePersistedViewState`) so concurrent provider instances merge without lost updates; `viewStates` is pruned to the newest 50 entries (missing `updatedAt` sorts oldest).
- `setViewStateId` sanitizes ids and rejects `__proto__`: a per-view entry must never be keyable through the `Object.prototype` setter. The fresh-read guard treats a corrupted non-object storage value as an empty map.
- View-local buffer (`viewLocalState`): `mode` / `currentApiConfigName` / `apiConfiguration` (non-secret subset) live per view in memory. `saveViewState` awaits the durable write before logging success. `loadViewState` hydrates at registration, keeps the profile name and logs when the profile lookup fails, and discards a stale load when the `viewStateId` changes mid-lookup.
- `setValues` / `setValue` validate `mode` against `getModeBySlug` (unknown → log and ignore; non-string passes through) and keep or clear the matching buffer fields; `undefined` / `null` values delete the buffer field rather than storing it. `getValues` merges context values with the buffer (buffer wins).
- `postMessageToWebview` no longer awaits the webview ack (a remounted or disposed page never acknowledges; awaiting would wedge task-critical callers).
- `resetState` clears `viewLocalState` and the view's persisted entry (after the `customModesManager.resetCustomModes` modal confirm).
- History restore (sticky-mode spec) writes the restored mode view-locally via `saveViewState("mode", ...)` instead of the shared global mode.

## Measurements
- a+d vs upstream/main @ 0d937c050: **999** (976+/23−) — over the 400 soft budget; measured at cut (`git diff --numstat 0d937c050..HEAD`); under the 1000 hard cap. Composition: impl + types + adapted history-restore tests ≈ 424 a+d (ClineProvider.ts 391, sticky-mode spec 15, packages/types 16, suppressions 2); the remainder is the new `view state persistence edge cases` describe (17 focused tests) plus spec fixture adaptation.
- src executable lines (mutation preflight): ClineProvider.ts 391 a+d (382+/9−) — under the 500-line cap; the gate run produced 179 raw mutants (under the 400 cap).

## Gates
- eslint --prune-suppressions: pass (suppression counts unchanged: ClineProvider.spec.ts `no-explicit-any` 198; prune-only reindent reverted)
- check-types: pass (11 packages)
- vitest: ClineProvider.spec.ts + ClineProvider.sticky-mode.spec.ts 204 pass
- stryker-diff ci @ 0d937c050: **179/179 killed, 0 surviving, 0 uncovered, 0 blocking** (ClineProvider.ts)
- e2e / i18n / visual: n/a (zero new i18n strings; no webview-ui changes)

## Parked / documented
From the gap-review parked-items register (F1a scope, all bounded):
1. Dev/prod `viewStateId` divergence — inherent to the browser mock.
2. Pre-launch write-queue / rekey orphan window + cross-session temp-id collision — narrow window, prune-bounded, non-secret.
3. Cross-process write-queue interleave — pre-existing memento semantics.
4. Redundant repoint branch — cosmetic.
5. Launch-repair queue sync — transient, self-healing.
6. Same-provider two-writer interleave test — cross-instance interleave already covered.
7. Flat-mutation apiConfiguration replace — coherent via the `getState()` re-merge (lands F1b).

## Porting notes
All F1a content is re-implemented against the base by hand-porting hunks from CS `e9a44b2fa` (#977 head): the durable core (`getPersistedViewStates` fresh-read guard, `savePersistedViewState` queued merge + prune, `clearPersistedViewState`, `prunePersistedViewStates`, `rekeyPersistedViewStateEntry`, `setViewStateId`, `loadViewState`, `saveViewState`), the view-local buffer with the `setValue` / `setValues` / `getValues` mutation handlers, the `postMessageToWebview` void-ack, the `resetState` clear, the `viewStates` record in `GLOBAL_STATE_KEYS` + types (global-settings.ts / vscode-extension-host.ts / index.test.ts), and the two history-restore tests adapted in ClineProvider.sticky-mode.spec.ts. The CS F1 spec (1790-line `parallelMode.spec.ts`) is NOT ported as one file: the F1-series describes are rewritten into the existing `ClineProvider.spec.ts` fixture (drops the 588-line mock preamble).
- Only intentional delta from CS: `setViewStateId` gains the 5-line `__proto__` rejection (A1 review hardening); the CS `stryker-ignore` comment is dropped — the guard is covered by the mutation gate instead.
- The six-item CS-hunks-not-ported register is observed (kimi-code OAuth try/catch; ApiConfigManager className tweak; ApiConfigManager.visual.tsx deletion + baselines; mojibake comment hunk; unused `defaultModeSlug` import — F3 re-adds it; providers/* + repo-config churn) — none ported here.

### CODE REVIEW

**Comment by easonLiangWorldedtech** [comment]
## CI status update (2026-09-14)

Merged the latest main (`ba46d1f34`) to advance the merge-base. The previous `mutation-diff` failure was base drift, not a defect in this PR: the gate's diff (merge-base → PR merge-ref) included ~a week of main's own changed lines, and main's code is not covered by this PR's tests. Upstream #1610 has also made surviving mutants advisory since.

The diff now contains only this PR's own changes; expecting a green re-run of the mutation gate on the new head.

File: src/core/webview/ClineProvider.ts [hunk 1]
```diff
@@ -1948,11 +2325,43 @@ export class ClineProvider
...
-		await this.contextProxy.setValues({
-			...globalSettings,
-			currentApiConfigName: profileToActivate,
-			listApiConfigMeta: entries,
-		})
+		// Write only the profile list back: replaying the full settings snapshot
+		// captured above would also rewrite unrelated keys (including viewStates,
+		// which ClineProvider mutates directly in storage for concurrent views)
+		// with this view's stale cached copy.
+		await this.contextProxy.setValue("listApiConfigMeta", entries)
```
[the remark this answers is not in the document]
Comment (easonLiangWorldedtech) [reply src/core/webview/ClineProvider.ts:2391]: Addressed in c9e5aa907 (now in this PR's own layer — previously only in the F1b layer of the stack): deleteProviderProfile now calls providerSettingsManager.deleteConfig(profileToDelete.name) after the last-profile guard and before persisting the updated listApiConfigMeta metadata, so the profile is removed from the persisted apiConfigs too and a later listConfig()/listApiConfigMeta refresh cannot restore it. The call branches on the typed ProviderSettingsNotFoundError (introduced in this PR by ProviderSettingsManager.deleteConfig), so an already-gone secret is an idempotent success — the stale list entry is still pruned — while any other failure (e.g. the store refusing to delete the last remaining configuration) propagates. Matching message text instead would let a profile whose name contains "not found" swallow an unrelated failure. Regression tests: the dangling-mode-mapping resurrection scenario ("removes the stored profile so a dangling mode mapping can no longer re-activate it"), the already-gone secret, and the store-level last-profile refusal in ClineProvider.sticky-profile.spec.ts; the typed-signal contract in ProviderSettingsManager.spec.ts; and provider-level not-found/propagation pins in ClineProvider.spec.ts.

File: src/core/webview/ClineProvider.ts [hunk 2]
```diff
@@ -1736,6 +2080,14 @@ export class ClineProvider
 	): Promise<void> {
 		const task = targetTask
 
+		// A cancelled or timed-out switch must not be partially applied: bail out
+		// before the task history / _taskMode writes as well as the durable mode
+		// write below. The pre-write check further down still covers aborts that
+		// land while the task writes are in flight.
+		if (signal?.aborted) {
+			return
+		}
```
[the remark this answers is not in the document]
Comment (easonLiangWorldedtech) [reply src/core/webview/ClineProvider.ts:2090]: Addressed in c94b42781: the in-flight window is now closed. If the mutation signal aborts while the task-history write is still pending, handleModeSwitchUnlocked rolls back the landed history write (re-writing the pre-switch item) and returns before the TaskModeSwitched emission and before the _taskMode assignment, so the task instance, task history, and provider state cannot retain the new mode. TaskModeSwitched is emitted only after persistence has settled and the signal is still un-aborted, and a view with no history item aborts before any write as well. A new deterministic test (ClineProvider.sticky-mode.spec.ts, "rolls back the landed history write and leaves no partial mode state when the signal aborts in flight") defers a controlled updateTaskHistory() call, aborts while it is pending, and asserts the rollback restores the original item, _taskMode is unchanged, TaskModeSwitched is not emitted, and no durable mode write occurs.

File: src/core/webview/ClineProvider.ts [hunk 3]
```diff
@@ -1958,11 +2352,52 @@ export class ClineProvider
...
+			)
+		}
+
+		// Capture this view's pin before setValue rewrites it: a view pinned to the
+		// deleted profile while the global selection points elsewhere must still be
+		// reconfigured, or getState() would keep the deleted profile's settings under
+		// the surviving profile's name.
+		const viewWasPinnedToDeleted = this.viewLocalState.currentApiConfigName === profileToDelete.name
+
+		await this.setValue("currentApiConfigName", profileToActivate)
```
[the remark this answers is not in the document]
Comment (easonLiangWorldedtech) [reply src/core/webview/ClineProvider.ts]: Addressed in c94b42781: deleteProviderProfile now captures viewWasPinnedToDeleted and deletedWasGlobal before any rewrite and makes the pin update conditional. setValue("currentApiConfigName", profileToActivate) — which updates both the view-local buffer and the shared store — runs only when this view is pinned to the deleted profile; when the deleted profile was the global selection but the view is pinned elsewhere, only the shared store is updated via contextProxy.setValue, so an unrelated local pin is preserved. The existing apiConfiguration reconfiguration guard is unchanged. The keeper test now seeds the view with a local pin (and nested buffer entry) on a different profile and asserts getValues() still reports the local pin while the shared store has moved to the surviving selection.
